Public/Invoke-Locksmith.ps1

```diff
@@ -162,24 +186,42 @@
         $CAHosts | ForEach-Object { $SafeUsers += '|' + $_.Name }
     }
 
-    Write-Host 'Identifying auditing issues...'
-    [array]$AuditingIssues = Find-AuditingIssue -ADCSObjects $ADCSObjects | Sort-Object Name
-
-    Write-Host 'Identifying AD CS templates with dangerous configurations...'
-    [array]$ESC1 = Find-ESC1 -ADCSObjects $ADCSObjects -SafeUsers $SafeUsers
-    [array]$ESC2 = Find-ESC2 -ADCSObjects $ADCSObjects -SafeUsers $SafeUsers
-    [array]$ESC3 = Find-ESC3Condition1 -ADCSObjects $ADCSObjects -SafeUsers $SafeUsers
-    [array]$ESC3 += Find-ESC3Condition2 -ADCSObjects $ADCSObjects -SafeUsers $SafeUsers
-
-    Write-Host 'Identifying AD CS template and other objects with poor access control...'
-    [array]$ESC4 = Find-ESC4 -ADCSObjects $ADCSObjects -SafeUsers $SafeUsers -DangerousRights $DangerousRights -SafeOwners $SafeOwners | Sort-Object Name
-    [array]$ESC5 = Find-ESC5 -ADCSObjects $ADCSObjects -SafeUsers $SafeUsers -DangerousRights $DangerousRights -SafeOwners $SafeOwners | Sort-Object Name
-    [array]$ESC6 = Find-ESC6 -ADCSObjects $ADCSObjects | Sort-Object Name
+    if ( $Scans ) {
+    # If the Scans parameter was used, Invoke-Scans with the specified checks.
+        $Results = Invoke-Scans -Scans $Scans
+            # Re-hydrate the findings arrays from the Results hash table
+            $AllIssues      = $Results['AllIssues']
+            $AuditingIssues = $Results['AuditingIssues']
+            $ESC1           = $Results['ESC1']
+            $ESC2           = $Results['ESC2']
+            $ESC3           = $Results['ESC3']
+            $ESC4           = $Results['ESC4']
+            $ESC5           = $Results['ESC5']
+            $ESC6           = $Results['ESC6']
+            $ESC8           = $Results['ESC8']
+    }
 
-    Write-Host 'Identifying HTTP-based certificate enrollment interfaces...'
-    [array]$ESC8 = Find-ESC8 -ADCSObjects $ADCSObjects | Sort-Object Name
+    if ( -not ($Scans) -and (-not ($Results)) ) {
```

Comment:
At this point, if we have tested and trust Invoke-Scans and its "rehydrated results" above, this block actually isn't needed anymore. The default value for the Scans parameter could just be set to 'All' and it would still process with Mode 0 exactly the same way.